Userland/Libraries/LibCore/AnonymousBuffer.cpp

```diff
@@ -11,6 +11,10 @@
 #include <fcntl.h>
 #include <sys/mman.h>
 
+#ifndef MAP_FILE
+#    define MAP_FILE 0
+#endif
```

Comment:
This feels very.. not great. Is this needed with the use of the temp shm files like on macOS/BSD?

Or does mmap(MAP_SHARED) with an fd parameter work on Haiku like MAP_FILE does on other platforms?

Reply:
`MAP_FILE` on most other platforms is a no-op.

Linux manpages:
>        MAP_FILE
>              Compatibility flag.  Ignored.

macOS manpages:
>      MAP_FILE           Mapped from a regular file or character-special device
>                        memory.  (This is the default mapping type, and need
>                        not be specified.)

FreeBSD manpages do not even note `MAP_FILE`, and in their headers it is defined to be `0`. The same is true on OpenBSD. The POSIX specification makes no mention of such a flag.

So, is there anywhere it is actually needed, or can it just be dropped altogether rather than adding this hack for Haiku?

Reply:
Sounds reasonable to me, I opened a PR to remove our uses of it outside of LibC/sys/mman.h. We weren't using it in the Kernel either. https://github.com/SerenityOS/serenity/pull/20885